__init__.py

```diff
@@ -661,11 +726,23 @@ def handle_stt_selected(self, message):
                     callback=handle_intent_aborted)
     def handle_tts_menu(self):
         self.state = SetupState.SELECTING_TTS
-        self.gui["offlineMale"] = self.setup.offline_male_tts_module
-        self.gui["onlineMale"] = self.setup.online_male_tts_module
-        self.gui["offlineFemale"] = self.setup.offline_female_tts_module
-        self.gui["onlineFemale"] = self.setup.online_female_tts_module
-        self.handle_display_manager("BackendLocalTTS")
+        model = get_tts_lang_configs(lang=self.selected_language, include_dialects=True)
+        supported_tts_engines = list()
+        for engine, configs in model.items():
+            # For Display purposes, we want to show the engine name without the underscore or dash and capitalized all
+            plugin_display_name = engine.replace("_", " ").replace("-", " ").title()
+            # Need to go one level deeper to get the voice name and gender
+            for voice in configs:
+                supported_tts_engines.append({"plugin_name": plugin_display_name,
+                                              "display_name": voice.get("display_name", "unknown"),
+                                              "gender": voice.get("gender", "unknown"),
+                                              "offline": voice["offline"],
+                                              'engine': engine})
+
+        # Randomize the order of supported TTS engines list to give a more natural feel to the selection
+        random.shuffle(supported_tts_engines)
```

Comment:
not sure this is good, i'd rather have determinism and proper sorting, maybe we need some sort of optional "priority" key in plugins (default 0) and sort based on that? this way i can make "shitty plugins" show at the bottom

Reply:
Unless the function itself is providing a sorted list, i am not sure there is anyways to track priority here

Reply:
i mean adding this new priority concept in OPM and in individual plugins, its not there yet, will be a new PR for me to work on if you like the idea

Reply:
Random shuffle has been removed for now but it makes the UI look ugly

Reply:
you can leave it until the sorting is added to OPM, this was a conversation starter not a change request, up to you

Reply:
lets wait on opm to do whatever it needs to